src/controller/PostController.php

```diff
@@ -71,40 +81,71 @@ public function __construct(TemplateInterface $template)
     public static function getInstance(TemplateInterface $template) :PostController
     { 
         if (is_null(self::$_instance)) {
-            self::$_instance = new PostController($template);  
+            self::$_instance = new PostController($template);
         }
     
         return self::$_instance;
     }
 
     /**
-     * Summary of getPosts
+     * Summary of showPosts
      * 
-     * @return array
+     * @return void
      */
-    public function getPosts() :array
+    public function showPosts() :void
     {
-        $postRepository = new PostRepository();
-        $results= $postRepository->getAllPostsWithAuthors();
+        $result = $this->_postService->getPosts();
+        echo $this->template->render($this::POSTS_VIEW, ['posts' => $result]);
+    }
 
-        $postmapper = new PostsMapper();
-        $posts = $postmapper->transformToListOfPostModel($results);
+    /**
+     * Summary of showPostDetails
+     * 
+     * @param int $postId id of the post
+     * 
+     * @return void
+     */
+    public function showPostDetails(int $postId) :void
+    {
+        $message = null;
+        if (isset($_POST["action"])) {
+            if ($_POST["action"] === $this->_commentService::ACTION) {
+                $isSubmitted = $this->isSubmitted($this->_commentService::ACTION);
+                $isValid = $this->isValid($_POST);
+                if ($isSubmitted && $isValid) {
+                    $comment = $this->_commentService->manageComment();
+                    $validateComment = $this->validCommentForm($comment);
 
-        return $posts;
+                    $message = $this->_commentService->createNewComment($validateComment);
+                } else {
+                    $message = [
+                        MessageService::ERROR => MessageService::GENERAL_ERROR
+                    ];
+                }
+            }
+        }
+        $postDetails = $this->_postService->getPostDetails($postId);
+        echo $this->template->render(
+            self::ONEPOST_VIEW, [
+                'id' => $postId,
```

Comment:
```suggestion
                'id' => $postDetails->getPostId() ....
```
Avec un objet celà serait mieux.

Reply:
fait